tests/test_unit/test_napari_plugin/test_poses_loader_widget.py

```diff
@@ -0,0 +1,170 @@
+"""Unit tests for loader widgets in the napari plugin.
+
+We instantiate the PosesLoader widget in each test instead of using a fixture.
+This is because mocking widget methods would not work after the widget is
+instantiated (the methods would have already been connected to signals).
+"""
+
+import pytest
+from napari.settings import get_settings
+from pytest import DATA_PATHS
+from qtpy.QtWidgets import QComboBox, QLineEdit, QPushButton, QSpinBox
+
+from movement.napari._loader_widgets import SUPPORTED_POSES_FILES, PosesLoader
+
+
+# ------------------- tests for widget instantiation--------------------------#
+def test_poses_loader_widget_instantiation(make_napari_viewer_proxy):
+    """Test that the loader widget is properly instantiated."""
+    # Instantiate the poses loader widget
+    poses_loader_widget = PosesLoader(make_napari_viewer_proxy)
+
+    # Check that the widget has the expected number of rows
+    assert poses_loader_widget.layout().rowCount() == 4
+
+    # Make sure the all rows except last start with lowercase text
+    # which ends with a semicolon
+    for i in range(poses_loader_widget.layout().rowCount() - 1):
+        label = poses_loader_widget.layout().itemAt(i, 0).widget()
+        assert label.text().islower()
+        assert label.text().endswith(":")
+
+    # Make sure that the source software combo box is populated
+    source_software_combo = poses_loader_widget.findChildren(QComboBox)[0]
+    assert source_software_combo.count() == len(SUPPORTED_POSES_FILES)
+
+    # Test that the default fps is 30
+    fps_spinbox = poses_loader_widget.findChildren(QSpinBox)[0]
+    assert fps_spinbox.value() == 30
+
+    # Make sure that the line edit for file path is empty
+    file_path_edit = poses_loader_widget.findChildren(QLineEdit)[-1]
+    assert file_path_edit.text() == ""
+
+    # Make sure that the first button is a "Browse" button
+    browse_button = poses_loader_widget.findChildren(QPushButton)[0]
+    assert browse_button.text() == "Browse"
+
+    # Make sure that the last row is a "Load" button
+    load_button = poses_loader_widget.findChildren(QPushButton)[-1]
+    assert load_button.text() == "Load"
+
+    # Make sure that layer tooltips are enabled
+    assert get_settings().appearance.layer_tooltip_visibility is True
+
+
+# --------test connection between widget buttons and methods------------------#
+def test_browse_button_calls_on_browse_clicked(
+    make_napari_viewer_proxy, mocker
+):
+    """Test that clicking the 'Browse' button calls the right function."""
+    mock_method = mocker.patch(
+        "movement.napari._loader_widgets.PosesLoader._on_browse_clicked"
+    )
+    poses_loader_widget = PosesLoader(make_napari_viewer_proxy)
+    browse_button = poses_loader_widget.findChildren(QPushButton)[0]
```

Comment:
If you adopt the above suggestion and have names, we can use `findChild` here instead